sljit_src/sljitNativeMIPS_common.c

```diff
@@ -353,6 +366,60 @@ static sljit_s32 function_check_is_freg(struct sljit_compiler *compiler, sljit_s
 
 #endif /* SLJIT_CONFIG_MIPS_32 && SLJIT_ARGUMENT_CHECKS */
 
+static void get_cpu_features(void)
+{
+#if defined(__GNUC__)
+#if !defined(SLJIT_IS_FPU_AVAILABLE)
+	sljit_u32 fir;
+#endif
+#if defined(SLJIT_CONFIG_MIPS_32) && SLJIT_CONFIG_MIPS_32
+#if (!defined(SLJIT_DETECT_FR) || (defined(SLJIT_DETECT_FR) && SLJIT_DETECT_FR == 1)) \
+	&& (!defined(SLJIT_DETECT_FR) || SLJIT_DETECT_FR == 1) \
+	&& (!defined(SLJIT_MIPS_REV) || (defined(SLJIT_MIPS_REV) && SLJIT_MIPS_REV < 6))
+	sljit_sw flag = -1;
+#endif
+#endif /* SLJIT_CONFIG_MIPS_32 */
+#endif /* __GNUC__ */
+	int feature_list = CPU_FEATURE_DETECTED;
+
+#if defined(__GNUC__) && !defined(SLJIT_IS_FPU_AVAILABLE)
+	__asm__ ("cfc1 %0, $0" : "=r"(fir));
+#endif
+	if ((fir & (1 << 22)) != 0)
+		feature_list |= CPU_FEATURE_FPU;
+
+	if (!(feature_list & CPU_FEATURE_FPU) &&
+		(fir & (1 << 17)) != 0 && (fir & (1 << 21)) != 0)
```

Comment:
since the bits are not adjacent, it is easier to see which ones are relevant when looking at the documentation this way.

frankly I suspect that only one of them is required for our use case, but since I have no mips32r1 CPU to test, this is safer.